debbuild

```diff
@@ -1812,181 +1983,522 @@ sub store_value {
+    return "%{$macro $params}";
+  }
+  $s =~ s/(?:\A|\R)\s*?%(define|undefine|global)\s+(\S+)\s*((?:.*\\\n)*.*|.*)/store_value($1,$2,$3);''/eg;
+  $s =~ s/(\A|\R[ \t]*)%([a-zA-Z_]\w+)[ \t]+(.*)/$1.expd_tocomplex($2, $3)/eg;
+  my $l = length $s;
+  sub expd_simple {
+    my $m = shift;
+    my $o = $m;
+    my $macroparms = {'#' => 0};
+    foreach my $ms ( @macropsstk ) {
+      $macroparms = $ms;
+      last if $macroparms->{'#'};
     }
+    if ( $m =~ /\A(\d+|#|\*{1,2})\z/ ) {
+      $m = defined($macroparms->{$1}) ? $macroparms->{$1} : '';
+    } elsif ( $m eq 'dump' ) {
+      $m = dump_macros();
+    } elsif ( $m eq 'optflags' ) {
+      $m = ($pkgdata{main}{arch} and $optflags{$pkgdata{main}{arch}}) ?
+           $optflags{$pkgdata{main}{arch}} : '';
+    } elsif ( $m eq 'getconfdir' ) {
+      $m = $specglobals{_prefix}."/lib/debbuild";
+    } elsif ( $m eq 'sources' ) {
+      $m = join ' ', sources();
+    } elsif ( $m eq 'patches' ) {
+      $m = join ' ', sources();
+    } elsif ( $m eq 'source' ) {
+      $m = "%{_sourcedir}/".$pkgdata{main}{source};
+    } elsif ( $m =~ /^source(\d+)/i ) {
+      $m = defined($pkgdata{sources}{$1}) ? "%{_sourcedir}/".$pkgdata{sources}{$1} : '';
+    } elsif ( $m =~ /^(patch\d+)/i ) {
+      $m = defined($pkgdata{main}{$1}) ? "%{_sourcedir}/".$pkgdata{main}{$1} : '';
+    } elsif ( defined($specglobals{$m}) ) {
+      my $macro = $m;
+      $m = $specglobals{$m};
+      if ( $m =~ /\A%\{lua:/ ) {
+        my $tst = $m;
+        $tst =~ s/\A%//;
+        if ( defined(extract_bracketed($tst, '{}')) ) {
+          $m =~ s/\A%\{lua:\s+//;
+          $m =~ s/\}\s*\z//;
+          my %macroparms = ();
+          $macroparms{0} = $macro;
+          $macroparms{'#'} = 0;
+          $macroparms{'**'} = $macroparms{'*'} = '';
+          unshift @macropsstk, \%macroparms;
+          $m = lua_macro($m);
+          shift @macropsstk;
+        }
+      }
+    } elsif ( defined($pkgdata{main}{$m}) ) {
+      $m = $pkgdata{main}{$m};
+    } else {
+      $m = '%'.$m;
+    }
+    vdebug($o, 5, "expandmacros/simple/in");
+    my $i = scalar @macropsstk;
+    vdebug("Parameterized macro stack: [".($i)."]", 7, "expandmacros/simple/stack");
+    foreach my $ms ( @macropsstk ) {
+      vdebug("[$i] ".$ms->{'0'}." / ".$ms->{'#'}.": ".$ms->{'**'}, 7, "expandmacros/simple/stack");
+      $i--;
+    }
+    vdebug($m, 5, "expandmacros/simple/out");
+    return ($m eq $o or $m eq '%'.$o) ? $m : expandmacros($m);
   }
-  pos = 0; # reset to start of $_
-
-  # Expand 'simple' '%{macros}': %define's and %{all_the_rest}
-  while (/%\{([a-z_]\w*)}/ig) {
-    my $key = $1;
-    if (defined $specglobals{$key}) {
-      s/%\{$key}/expandmacros($specglobals{$key})/eg;
-    } elsif (defined $pkgdata{main}{$key}) {
-      s/%\{$key}/expandmacros($pkgdata{main}{$key})/eg;
+  sub expd_complex {
+    my $m = shift;
+    my $o = $m;
+    my ($qex, $macro, $value);
```

Comment:
Not really sure about `$m` there. It's used to many times within `expandmacros`, but with the same meaning. Maybe just add commect what is it? Otherwise the size of the script will be increased, but not really adds readability.